tests/cephadm/test_bootstrap.py

```diff
@@ -399,6 +456,64 @@ def validate_skip_dashboard(cls, flag, response):
     log.info("skip-dashboard validation is successful")
 
 
+def validate_ssh_public_key(cls, ssh_public_key):
+    """
+    Method to validate ssh public key
+    ssh_public_key:
+        if specified, then the specified public key should be configured for the cluster
+    Args:
+        cls: cephadm instance
+        ssh_public_key: ssh-public-key bootstrap option
+        response: bootstrap command response
+    """
+    out, _ = cls.shell(args=["ceph", "cephadm", "get-pub-key"])
+    if ssh_public_key and file_or_path_exists(cls.installer, ssh_public_key):
+        pub_key = fetch_file_content(cls.installer, ssh_public_key)
+
+    log.info(
+        f"ssh-public-key configured: {out}, ssh-public-key used in cluster: {pub_key}"
+    )
+
+    # if pub_key found and out == pub_key, configured and used keys are same. Valid
+    # if pub_key found and out != pub_key, configured and used keys are different, not Valid
+    if pub_key and out != pub_key:
+        raise BootStrapValidationFailure(
+            "ssh public key provided in the configuration is not used in the cluster"
+        )
+
+    log.info("ssh-public-key validation is successful")
+
+
+def validate_ssh_private_key(cls, ssh_private_key):
+    """
+    Method to validate ssh public key
+    ssh_public_key:
+        if specified, then the specified public key should be configured for the cluster
+    Args:
+        cls: cephadm instance
+        ssh_private_key: ssh-private-key bootstrap option
+        response: bootstrap command response
+    """
+    out, _ = cls.shell(
+        args=["ceph", "config-key", "get", "mgr/cephadm/ssh_identity_key"]
+    )
+    if ssh_private_key and file_or_path_exists(cls.installer, ssh_private_key):
```

Comment:
I think file existence checked in one of the method `verify_ceph_custom_public_private_ssh_key`.

Reply:
do we need this?

Reply:
Added file existence check in validate method and removed the verify method for ssh keys